### DISCUSSION TITLE
ORY Hydra -vs- Kratos

### DISCUSSION THREAD

**Discussion by mnyv** [ory/kratos#1104] [Q&A] [answered]
Hi

I recently wrote a small POC to try ORY Hydra. It works well but the flow is fairly complicated. I like being able to design the UI as I want. It's big step up from trying to theme FreeMarker templates for example. But of course on the flipside, I have to manually set up password reset, email verification and so on. 

I'm considering trying ORY Kratos, but I don't understand where is fits in and how it works. 

Can you tell me if Kratos would be a good match for this use case, for example?

---

Here's the setup:

I currently have two Single Page Applications written in Angular and a REST API written in Spring. The SPA:s use authentication to determine weather or not users can navigate to certain routes and uses roles to determine what users see on those routes. 

Similarly, the API uses authentication/roles to determine who can do what in the system. 

Very likely other sites will be added in the future on the same domain (x.domain.com, y.domain.com, etc) and logging in with a single id / pass should be enough to access all of them.

**Comment by Benehiko** [comment, marked as answer]
Hi @mnyv,

Ory Hydra is for allowing external parties to access your users' data (think of "sign in with Google" on app/website X). Ory Kratos is solving user registration/login/recovery flows which makes this different from Ory Hydra as it relates to your users interacting directly with your system instead of via a 3rd party. Please see this documentation https://www.ory.sh/hydra/docs/concepts/before-oauth2 and https://www.ory.sh/kratos/docs/concepts/index

In your case you want SSO (single-sign on) which requires a session across sub-domains - Ory Kratos can do that :)

**Reply by mnyv** [reply]
Thanks @Benehiko

**Comment by vinckr** [comment]
Hey @mnyv 
The Ory Kratos and Ory Hydra integration just landed with Hydra 2.0, 
see the [release notes](https://github.com/ory/hydra/releases/tag/v2.0.0) for more details!
